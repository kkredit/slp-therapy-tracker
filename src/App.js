import React, { Component } from 'react'
import Container from 'react-bootstrap/Container'
import { BrowserRouter, Route } from 'react-router-dom'

import Includes from './Includes.js'
import Navbar from './Navbar.js'
import Home from './Home.js'
import Providers from './Providers.js'
import Locations from './Locations.js'
import History from './History.js'
import Session from './Session.js'

// import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Container>
          <Includes />
          <BrowserRouter>
            <Navbar />

            <Route exact path="/" component={Home} />
            <Route exact path="/providers" component={Providers} />
            <Route exact path="/locations" component={Locations} />
            <Route exact path="/history" component={History} />
            <Route exact path="/session" component={Session} />
          </BrowserRouter>
        </Container>
      </div>
    );
  }
}

export default App;
