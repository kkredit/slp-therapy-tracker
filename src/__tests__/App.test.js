import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { create } from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Snapshots',()=>{
  test('Loading screen', () => {
    let tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })

  test('Home screen', () => {
    let tree = create(<App />)
    let instance = tree.getInstance();
    instance.dbInitialLoadCb();
    expect(tree.toJSON()).toMatchSnapshot();
  })
})

describe("Loading screen should give way to the app", () => {

  test('Simulate loading the db', () => {
    let tree = create(<App />);
    let instance = tree.getInstance();
    expect(tree.toJSON()).toMatchSnapshot();

    expect(instance.state.dbLoaded).toBe(false);
    instance.dbInitialLoadCb();
    expect(instance.state.dbLoaded).toBe(true);

    expect(tree.toJSON()).toMatchSnapshot();
  })
})
