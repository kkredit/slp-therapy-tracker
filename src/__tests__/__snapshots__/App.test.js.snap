// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading screen should give way to the app Simulate loading the db 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Loading...
    </p>
  </header>
</div>
`;

exports[`Loading screen should give way to the app Simulate loading the db 2`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <link
      crossOrigin="anonymous"
      href="https://stackpath.bootstrapcdn.com/bootswatch/4.3.1/yeti/bootstrap.min.css"
      integrity="sha384-w6tc0TXjTUnYHwVwGgnYyV12wbRoJQo9iMlC2KdkdmVvntGgzT9jvqNEF/uKaF4m"
      rel="stylesheet"
    />
    <link
      crossOrigin="anonymous"
      href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css"
      integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN"
      rel="stylesheet"
    />
    <div
      className="Navbar nav nav-pills"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/history"
          onClick={[Function]}
        >
          History
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/session"
          onClick={[Function]}
        >
          Session
        </a>
      </div>
    </div>
    <h1>
      SLP Therapy Tracker
    </h1>
    <a
      className="btn btn-primary btn-lg"
      href="/session"
      id="mainButton"
      onClick={[Function]}
    >
      Start a session 
      <i
        className="fa fa-arrow-circle-o-right"
      />
    </a>
    <p>
      This app is designed to ease data collection for Speech and Language Pathologists. Made with &lt;3 for Kristen.
    </p>
    <h3
      className="subheader"
    >
      How to Use
    </h3>
    <ol>
      <li>
        Enter your data in the "Providers" form
      </li>
      <li>
        Enter service location names in the "Locations" form
      </li>
      <li>
        Enter session data in the "Session" page
      </li>
      <li>
        View historical session data in the "History" page
      </li>
    </ol>
    <a
      className="navButton btn btn-outline-primary"
      href="/providers"
      onClick={[Function]}
    >
      Update Providers 
      <i
        className="fa fa-arrow-circle-right"
      />
    </a>
    <a
      className="navButton btn btn-outline-primary"
      href="/locations"
      onClick={[Function]}
    >
      Update Locations 
      <i
        className="fa fa-arrow-circle-right"
      />
    </a>
    <h3
      className="subheader"
    >
      More Info
    </h3>
    <p>
      This app was made as a class project. See more info at the developer page.
    </p>
    <a
      className="navButton btn btn-outline-primary"
      href="/developer"
      onClick={[Function]}
    >
      Developer page
    </a>
    <h3
      className="subheader"
    >
      Disclaimer
    </h3>
    <p>
      This app is a class project. There is no authentication. The user is solely responsible for all information that is uploaded to this site.
    </p>
  </div>
</div>
`;

exports[`Snapshots Home screen 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <link
      crossOrigin="anonymous"
      href="https://stackpath.bootstrapcdn.com/bootswatch/4.3.1/yeti/bootstrap.min.css"
      integrity="sha384-w6tc0TXjTUnYHwVwGgnYyV12wbRoJQo9iMlC2KdkdmVvntGgzT9jvqNEF/uKaF4m"
      rel="stylesheet"
    />
    <link
      crossOrigin="anonymous"
      href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css"
      integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN"
      rel="stylesheet"
    />
    <div
      className="Navbar nav nav-pills"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/history"
          onClick={[Function]}
        >
          History
        </a>
      </div>
      <div
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/session"
          onClick={[Function]}
        >
          Session
        </a>
      </div>
    </div>
    <h1>
      SLP Therapy Tracker
    </h1>
    <a
      className="btn btn-primary btn-lg"
      href="/session"
      id="mainButton"
      onClick={[Function]}
    >
      Start a session 
      <i
        className="fa fa-arrow-circle-o-right"
      />
    </a>
    <p>
      This app is designed to ease data collection for Speech and Language Pathologists. Made with &lt;3 for Kristen.
    </p>
    <h3
      className="subheader"
    >
      How to Use
    </h3>
    <ol>
      <li>
        Enter your data in the "Providers" form
      </li>
      <li>
        Enter service location names in the "Locations" form
      </li>
      <li>
        Enter session data in the "Session" page
      </li>
      <li>
        View historical session data in the "History" page
      </li>
    </ol>
    <a
      className="navButton btn btn-outline-primary"
      href="/providers"
      onClick={[Function]}
    >
      Update Providers 
      <i
        className="fa fa-arrow-circle-right"
      />
    </a>
    <a
      className="navButton btn btn-outline-primary"
      href="/locations"
      onClick={[Function]}
    >
      Update Locations 
      <i
        className="fa fa-arrow-circle-right"
      />
    </a>
    <h3
      className="subheader"
    >
      More Info
    </h3>
    <p>
      This app was made as a class project. See more info at the developer page.
    </p>
    <a
      className="navButton btn btn-outline-primary"
      href="/developer"
      onClick={[Function]}
    >
      Developer page
    </a>
    <h3
      className="subheader"
    >
      Disclaimer
    </h3>
    <p>
      This app is a class project. There is no authentication. The user is solely responsible for all information that is uploaded to this site.
    </p>
  </div>
</div>
`;

exports[`Snapshots Loading screen 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Loading...
    </p>
  </header>
</div>
`;
